src/SolverAlsWrmf.cpp

```diff
@@ -304,11 +304,21 @@ double als_explicit_float(const Rcpp::S4 &m_csc_r,
 }
 
 template <class T>
-void initialize_biases(const dMappedCSC& ConfCSC,
-                       const dMappedCSC& ConfCSR,
-                       arma::Col<T>& user_bias,
-                       arma::Col<T>& item_bias,
-                       T lambda, bool non_negative) {
+double initialize_biases(const dMappedCSC& ConfCSC,
+                         const dMappedCSC& ConfCSR,
+                         arma::Col<T>& user_bias,
+                         arma::Col<T>& item_bias,
+                         T lambda, bool non_negative) {
+  /* Robust mean calculation */
```

Comment:
Where can I get more information about "Robust mean". Why is this better than just standard `mean`?

Reply:
Here's some discussion: https://stackoverflow.com/questions/7552443/whats-the-numerically-best-way-to-calculate-the-average

It gets a mean estimation which is correct to higher numerical precision that `sum(x)/n`. But it's not better than R's `mean` function if that's what you mean to ask (which uses a compensated summation).

In this case the sum is expected to be over a potentially very large number of elements, so a simpler mean calculation can lose precision.